ArduCopter/landing_gear.cpp

```diff
@@ -32,3 +47,16 @@ void Copter::landinggear_update()
 
     last_deploy_status = landinggear.deployed();
 }
+
+// True when copter is in a mode that should auto-deploy the landing gear
+bool Copter::landinggear_deploy_modes()
+{
+    if (control_mode == LAND ||
+       (control_mode == RTL && (rtl_state == RTL_LoiterAtHome || rtl_state == RTL_Land || rtl_state == RTL_FinalDescent)) ||
+       (control_mode == AUTO && auto_mode == Auto_Land) ||
+       (control_mode == AUTO && auto_mode == Auto_RTL && (rtl_state == RTL_LoiterAtHome || rtl_state == RTL_Land || rtl_state == RTL_FinalDescent))){
+        return true;
```

Comment:
This is a mouthful. Can we change it to switch(control_mode)?

Reply:
@magicrub As it turns out, there really isn't a better way to write this that I can think of. There's only three control modes. It's the "sub modes" that are wordy, and there isn't a way to make that not wordy. It just is.